### ISSUE TITLE
PrecisionAlignment.Found non–error?

### ISSUE THREAD

**Issue by Rarst** [WordPress-Coding-Standards/WordPress-Coding-Standards#1304]
I've started to get `WordPress.WhiteSpace.PrecisionAlignment.Found: Found precision alignment of 1 spaces` on code like this:

```php
$wp_admin_bar = $this->getMockBuilder( 'WP_Admin_Bar' )
                     ->allowMockingUnknownTypes()
                     ->setMethods( [ 'add_node' ] )
                     ->getMock();
```

I don’t think it is against the standard (neither does the message say so), so what’s the purpose of having this? Unless some action is expected about it (which it doesn't seem to be), it just clutters results?

_update_ Ok, if I disable this sniff then I get a _different_ error in the same place: `Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed`. So it seems former "masks" the latter? Is it intended behaviour if so?

**Comment by jrfnl** [comment]
> I don’t think it is against the standard (neither does the message say so), so what’s the purpose of having this? Unless some action is expected about it (which it doesn't seem to be), it just clutters results?

See #983 and #1099 which discuss the reasons for introducing the sniff in detail.

The short of it is that in WP Core, most such space usage was unintentional and not some form of precision alignment. This sniff allowed to detect the unintentionally incorrectly indented lines.

As the sniff can not distinguish between intentional precision alignment and unintentional incorrect indentation, the sniff throws a `warning` - in contrast to an error - and does not auto-fix.

If you prefer not to see these notices, you can always (selectively) exclude them a custom ruleset.

> So it seems former "masks" the latter? Is it intended behaviour if so?

Just ran the `WordPress` standard over your snippet and no masking is going on. Both the error from the `Generic.WhiteSpace.DisallowSpaceIndent` sniff as well as the warning from the `WordPress.WhiteSpace.PrecisionAlignment` sniff are being thrown.

The sniffs function independently of each other, so I was surprised by your remark and am glad to confirm no interference is going on.

I do wonder why you were not seeing both, but that might be something in your setup ?

**Comment by Rarst** [comment]
Sorry, I've found a bit about it in issues, but missed the PR.

Think the overlap is on PhpStorm display side, I see both sniffs if I run from command line.
